src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1065 @@
+                            CORRECT_BIT_RANGE_PUNCTUATION
+                        )
+                    ),
+                    Some(token_tree) => macro_error!(
+                        token_tree,
+                        format!(
+                            "Incorrect bit range token. {}",
+                            CORRECT_BIT_RANGE_PUNCTUATION
+                        )
+                    ),
+                    None => {
+                        macro_error!(
+                            field_ident,
+                            format!("Missing bit range token. {}", CORRECT_BIT_RANGE_PUNCTUATION)
+                        )
+                    }
+                };
+                // Check and get end index in bit range (e.g. 2nd literal/4th character here `1..2`)
+                let end = match fields_iter.next() {
+                    Some(TokenTree::Literal(field_end_pos)) => {
+                        if let Ok(pos) = field_end_pos.to_string().parse::<u8>() {
+                            pos
+                        } else {
+                            macro_error!(
+                                field_end_pos,
+                                "Failed to parse token for bit range end index."
+                            );
+                        }
+                    }
+                    Some(token_tree) => {
+                        macro_error!(token_tree, "Incorrect token for bit range end index.")
+                    }
+                    None => macro_error!(field_ident, "Missing token for bit range end index."),
+                };
+
+                let (valid_start, valid_end) = (
+                    (0..end).contains(&start),
+                    (start..bits_len.len()).contains(&end),
+                );
+
+                // To avoid indenting the following section with
+                // `if valid_start && valid_end` we do this check instead.
+                if !(valid_start || valid_end) {
+                    macro_error!(
+                        field_ident,
+                        format!(
+                            "\
+                            Range of given bit field is invalid, `(0..end).contains(&start) && \
+                             (start..bits_len).contains(&end)` evaluated to `false` (it must \
+                             evaluate to `true`): `(0..{}).contains(&{}) && \
+                             ({}..{}).contains(&{})` => `{} && {}` => `false`",
+                            end,
+                            start,
+                            start,
+                            bits_len.len(),
+                            end,
+                            valid_start,
+                            valid_end
+                        )
+                    );
+                }
+                // Set display string
+                // TODO With 1 bit range defined in struct, print will not work
+                // correctly, fix that.
```

Comment:
Is it fixed?

Reply:
I believe so, will add a unit test to verify this.